pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -151,8 +151,15 @@ func (c *Cluster) Start() error {
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
 
-			if err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, false); err != nil {
-				logger.Errorf("failed to update mgr deployment %q. %v", resourceName, err)
+			ContIfUnhealthy := c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy
+
+			err := updateDeploymentAndWait(c.context, c.clusterInfo, d, config.MgrType, mgrConfig.DaemonID, c.spec.SkipUpgradeChecks, ContIfUnhealthy)
+			if err != nil {
+				if ContIfUnhealthy {
```

Comment:
Instead of a local variable I'd suggest to just use the original property. 
```suggestion
				if c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy {
```

Reply:
@travisn  wrote:
> Instead of a local variable I'd suggest to just use the original property.

I introduced the local variable because I needed to use the Property in two places, to reduce duplication.

Reply:
For readability, I've found that the code is more maintainable if we just use the fields directly from `c.spec` instead of variables. It's helpful to see everywhere that we use CR settings.

Reply:
@travisn  wrote:
> For readability, I've found that the code is more maintainable if we just use the fields directly from `c.spec` instead of variables. It's helpful to see everywhere that we use CR settings.

Changed as requested.